src/octoprint/static/js/app/viewmodels/terminal.js

```diff
@@ -277,6 +277,17 @@ $(function() {
             }
         };
 
+        self.terminalScrollEvent = function () {
```

Comment:
Can you debounce the execution of this event? 
You can use https://lodash.com/docs/4.17.5#debounce for this

Technical:
onScroll is a pretty intensive event that get triggers alot. Further more you're retrieving a property that forces a layout reflow. Other wise known as "[Layout trashing](http://kellegous.com/j/2013/01/26/layout-performance/)", In order not to affect performance of the browser, you only want calculate with this property once every X ms.

Some more info if you would like to know what causes layout trashing: https://gist.github.com/paulirish/5d52fb081b3570c81e3a

Reply:
Ah nice didn't realize this existed. 

Are you suggesting I do something like:
> self.terminalScrollEvent = _.debounce(function() { ... }, 100);

Any idea what a sane value for wait would be?

Reply:
Yes, that should be the code. I think 500ms would be a save value.

Also in addition, you might need to set options of debounce to `leading: true` and `trailing: true` because debounce will properly not immediately disable auto-scrolling causing it to jump back to the bottom before disabling it.

Reply:
Sorry

Rename debounce to [throttle](https://lodash.com/docs/4.17.5#throttle), that one is already set with both true

Reply:
Ah that works. Gave it a go and it looks like 250 is a tolerable value.